systemd/docker.service

```diff
@@ -9,9 +9,12 @@ Wants=network-online.target
 # the default is not to use systemd for cgroups because the delegate issues still
 # exists and systemd currently does not support the cgroup feature set required
 # for containers run by docker
-ExecStart=/usr/bin/dockerd
+ExecStart=/usr/bin/dockerd \
+    --containerd /run/containerd/containerd.sock \
+    --default-runtime containerd \
+    --add-runtime containerd=runc \
+    -H unix://
```

Comment:
Do we need to specify `-H unix://` here?  It'd be nicer if I could control the hosts via the `hosts` key in `daemon.json` instead of having to edit systemd unit files.  For instance if I wanted to enable listening on the TCP API.

Barring that, can we make this easier to override with a drop in file?  For instance the instructions here are somewhat straightforward: https://success.docker.com/article/how-do-i-enable-the-remote-api-for-dockerd.  But now with this change I need to remember to add all the containerd stuff in the drop in file override as well depending on what version of the engine I'm using.

Reply:
I think we can remove this, it shouldn't be needed

Reply:
@russellcardullo fyi just chatted with @tiborvass and he can help by making the changes to the engine to have these default values. PR should go to ee-engine 18.09 first because that's where we have the burning desire